### ISSUE TITLE
Derivation Paths in use

### ISSUE THREAD

**Issue by 3rdIteration** [panteam-net/pt.BTC#7]
Just looking at this and while it seems that you are using BIP39 compatible seeds, you seem to be doing derivation either in a non-standard way, or on a non-standard path.

Just wondering if you can provide some more info about how this is implemented?

**Comment by ArtjomP** [comment]
Salute! Thank you for your question!
It depends on what kind of wallet the user selects. 
The app supports two kinds of addresses derivation:
- One address: (mnemonic + password) = seed -> PrivateKey -> PublicKey -> OneAddress
- [BIP 44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki): (mnemonic + password) = seed -> PrivateKey -> PublicKey -> BIP44 -> PublicAddresses.

Our BIP44 address derivation generates the addresses as [here](https://github.com/dan-da/hd-wallet-addrs).

**Comment by 3rdIteration** [comment]
So for the single address mode, what exactly are you doing in terms of going from seed to Privatekey?

For example, I used the Android app to generate a single address wallet, it gives the following mnemonic:
`spatial stereo thrive reform shallow blouse minimum foster eagle game answer worth size stumble theme crater bounce stay extra duty man weather awesome search`
(No additional password added)

This produces the address: 1EaGSR7uWp2hok3jTtNypjUuV3G4YyMxgt

If you are following BIP39, then the mnemonic would produce the BIP39 seed of:
c8dce0e14123f379f5681761d606527e9c366ab9c00edb65ca6140b032dc8450210987d8998186ac53ace4030eb9bcc41e2f3220820aa823716aef0e4e12ed4d

and if I take what is normally the BIP39 privkey part of that, it produces a different address to your app.

**Comment by 3rdIteration** [comment]
For context, I'm basically trying to recover a wallet that was created in January 2019, so need to understand how the single address mode works. (which seems to be the only one that would have been available back then) 

Also curious what the mnemonic generation bug was that was fixed in v1.71.0.0?

**Comment by ArtjomP** [comment]
No, we don't use any derivation strategy for single address mode. The code (using [NBitcoin library](https://github.com/MetacoSA/NBitcoin)) looks like this:
```cs
using NBitcoin;

var mnemonic = "spatial stereo thrive reform shallow blouse minimum foster eagle game answer worth size stumble theme crater bounce stay extra duty man weather awesome search";
var password = String.Empty;

var mnemonicForGen = new Mnemonic(mnemonic);
var privateKey = mnemonicForGen.DeriveExtKey(password);
var publicKey = privateKey.Neuter();

var network = Network.Main;

var address = publicKey.ScriptPubKey.GetDestinationAddress(network).ToString();

Console.WriteLine(address);
```

We have a console app to this code. Please, let us know if you need it.

Unfortunately, it's hard to say what was the bug back then in 1.71.0.0.
I think it didn't affect the address generation.

**Comment by 3rdIteration** [comment]
Ah riteo, so it's basically treating the extended public key as a standard public key and going from there...

Have reproduced it, so thanks for the info :)
